internal/nodes/avi_model_evh_nodes.go

```diff
@@ -0,0 +1,1029 @@
+			PublishKeyToRestLayer(modelName, key, sharedQueue)
+		}
+	}
+	// Now remove the secret relationship
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).RemoveIngressSecretMappings(objname)
+	utils.AviLog.Infof("key: %s, removed ingress mapping for: %s", key, objname)
+
+	// Remove the hosts mapping for this ingress
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).DeleteIngToHostMapping(objname)
+
+	// remove hostpath mappings
+	updateHostPathCacheV2(namespace, objname, hostMap, nil)
+}
+
+//DeleteStaleData : delete pool, EVH VS and redirect policy which are present in the object store but no longer required.
+func DeleteStaleDataForEvh(routeIgrObj RouteIngressModel, key string, modelList *[]string, Storedhosts map[string]*objects.RouteIngrhost, hostsMap map[string]*objects.RouteIngrhost) {
+	utils.AviLog.Debugf("key: %s, msg: About to delete stale data EVH Stored hosts: %v, hosts map: %v", key, utils.Stringify(Storedhosts), utils.Stringify(hostsMap))
+	for host, hostData := range Storedhosts {
+		utils.AviLog.Debugf("host to del: %s, data : %s", host, utils.Stringify(hostData))
+		shardVsName := DeriveHostNameShardVSForEvh(host, key)
+
+		if shardVsName == "" {
+			// If we aren't able to derive the ShardVS name, we should return
+			return
+		}
+		modelName := lib.GetModelName(lib.GetTenant(), shardVsName)
+		found, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		if !found || aviModel == nil {
+			utils.AviLog.Warnf("key: %s, msg: model not found during delete: %s", key, modelName)
+			continue
+		}
+		// By default remove both redirect and fqdn. So if the host isn't transitioning, then we will remove both.
+		removeFqdn := true
+		removeRedir := true
+		currentData, ok := hostsMap[host]
+		utils.AviLog.Warnf("key: %s, hostsMap: %s", key, utils.Stringify(hostsMap))
+		// if route is transitioning from/to passthrough route, then always remove fqdn
+		if ok && hostData.SecurePolicy != lib.PolicyPass && currentData.SecurePolicy != lib.PolicyPass {
+			if currentData.InsecurePolicy == lib.PolicyRedirect {
+				removeRedir = false
+			}
+			utils.AviLog.Infof("key: %s, host: %s, currentData: %v", key, host, currentData)
+			removeFqdn = false
+		}
+		// Delete the pool corresponding to this host
+		if hostData.SecurePolicy == lib.PolicyEdgeTerm {
+			aviModel.(*AviObjectGraph).DeleteVsForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, removeFqdn, removeRedir, true)
+		}
+		if hostData.InsecurePolicy != lib.PolicyNone {
+			aviModel.(*AviObjectGraph).DeleteVsForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, removeFqdn, removeRedir, false)
+
+		}
+		changedModel := saveAviModel(modelName, aviModel.(*AviObjectGraph), key)
+		if !utils.HasElem(modelList, modelName) && changedModel {
+			*modelList = append(*modelList, modelName)
+		}
+	}
+}
+
+func DeriveHostNameShardVSForEvh(hostname string, key string) string {
```

Comment:
The reason we want to add the EVH tag is for migration cases?

Reply:
Yes, also for the cases where multiple AKO syncing to same controller with some EVH enabled.